pkg/controller/pv.go

```diff
@@ -86,17 +87,16 @@ func (r *reconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 		if !ok {
 			r.retryMutex.Lock()
 			if _, ok = r.retryMap[azVolumeName]; !ok {
-				r.retryMap[azVolumeName] = 0
-				numRetry = 0
+				var zero uint32
+				r.retryMap[azVolumeName] = &zero
+				numRetry = &zero
 			}
 			r.retryMutex.Unlock()
 		}
 
-		if numRetry <= maxRetry {
+		if *numRetry <= maxRetry {
```

Comment:
Since you're starting with a count of zero and incrementing after this check, I believe this should be `*numRetry < maxRetry` otherwise you'll retry one more time than the max.

Reply:
Good catch :) Done